internal/users/db/db.go

```diff
@@ -0,0 +1,282 @@
+// Package db handles transaction with an underlying database to store user and group information.
+package db
+
+import (
+	"context"
+	"database/sql"
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"syscall"
+
+	// sqlite3 driver.
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/ubuntu/authd/internal/fileutils"
+	"github.com/ubuntu/authd/internal/users/db/bbolt"
+	"github.com/ubuntu/authd/log"
+)
+
+var (
+	filename = "authd.sqlite3"
+	//go:embed sql/create_schema.sql
+	createSchema string
+)
+
+// Manager is an abstraction to interact with the database.
+type Manager struct {
+	db   *sql.DB
+	path string
+	mu   sync.RWMutex
+}
```

Comment:
I like it! Do you think we can do better or are you happy with the solution?

Reply:
I'm happy with it :)